[Automation] javac compile fails for org.apache.maven.plugins:maven-surefire-plugin:3.3.0

## javac compile on ubuntu-latest / JDK 25 / mode current-defaults

Failure kind: javac compile
OS: ubuntu-latest
JDK: 25
Native-image mode: current-defaults
Reproducer: `env GVM_TCK_LV=3.3.0 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean compileTestJava -Pcoordinates=org.apache.maven.plugins:maven-surefire-plugin:3.0.0`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/27101166930
Last 300 lines of the log (excerpt truncated to fit the GitHub issue size limit):
```console
;
                              ^
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:13: error: package org.apache.maven.plugin.logging does not exist
import org.apache.maven.plugin.logging.SystemStreamLog;
                                      ^
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:22: error: cannot find symbol
    void executeLoadsPluginHelpResource() throws MojoExecutionException {
                                                 ^
  symbol:   class MojoExecutionException
  location: class HelpMojoTest

/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:35: error: cannot find symbol
    private static final class RecordingLog extends SystemStreamLog {
                                                    ^
  symbol:   class SystemStreamLog
  location: class HelpMojoTest
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:25: error: cannot access AbstractMojo
        mojo.setLog(log);
            ^
  class file for org.apache.maven.plugin.AbstractMojo not found
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:39: error: method does not override or implement a method from a supertype
        @Override
        ^
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:44: error: method does not override or implement a method from a supertype
        @Override
        ^
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:49: error: method does not override or implement a method from a supertype
        @Override
        ^
8 errors
[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed; see the compiler output below.
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:25: error: cannot access AbstractMojo
          mojo.setLog(log);
              ^
    class file for org.apache.maven.plugin.AbstractMojo not found
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:22: error: cannot find symbol
      void executeLoadsPluginHelpResource() throws MojoExecutionException {
                                                   ^
    symbol:   class MojoExecutionException
    location: class HelpMojoTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:35: error: cannot find symbol
      private static final class RecordingLog extends SystemStreamLog {
                                                      ^
    symbol:   class SystemStreamLog
    location: class HelpMojoTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:39: error: method does not override or implement a method from a supertype
          @Override
          ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:44: error: method does not override or implement a method from a supertype
          @Override
          ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:49: error: method does not override or implement a method from a supertype
          @Override
          ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:12: error: package org.apache.maven.plugin does not exist
  import org.apache.maven.plugin.MojoExecutionException;
                                ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:13: error: package org.apache.maven.plugin.logging does not exist
  import org.apache.maven.plugin.logging.SystemStreamLog;

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
12 actionable tasks: 2 executed, 10 up-to-date
                                        ^
  8 errors

* Try:
> Check your code and dependencies to fix the compilation error(s)
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 1s

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for org.apache.maven.plugins:maven-surefire-plugin:3.0.0 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 3s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 3.3.0 at 2026-06-07T18:50:19Z with exit code 1 after 4s
❌ ERROR: 'native-image run' for 3.3.0 failed with exit code 1.
SOFT-FAIL[native-image run][3.3.0][env GVM_TCK_LV=3.3.0 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean test -Pcoordinates=org.apache.maven.plugins:maven-surefire-plugin:3.0.0]
========================================================================================
 org.apache.maven.plugins:maven-surefire-plugin:3.0.0:3.3.0 stage javac compile
========================================================================================
Starting stage 'javac compile' for 3.3.0 at 2026-06-07T18:50:19Z
Command: env GVM_TCK_LV=3.3.0 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean compileTestJava -Pcoordinates=org.apache.maven.plugins:maven-surefire-plugin:3.0.0
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 772ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava FAILED
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE

> Task :compileTestJava FAILED
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:12: error: package org.apache.maven.plugin does not exist
import org.apache.maven.plugin.MojoExecutionException;
                              ^
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:13: error: package org.apache.maven.plugin.logging does not exist
import org.apache.maven.plugin.logging.SystemStreamLog;
                                      ^
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:22: error: cannot find symbol
    void executeLoadsPluginHelpResource() throws MojoExecutionException {
                                                 ^
  symbol:   class MojoExecutionException
  location: class HelpMojoTest
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:35: error: cannot find symbol
    private static final class RecordingLog extends SystemStreamLog {
                                                    ^
  symbol:   class SystemStreamLog
  location: class HelpMojoTest
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:25: error: cannot access AbstractMojo
        mojo.setLog(log);
            ^
  class file for org.apache.maven.plugin.AbstractMojo not found
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:39: error: method does not override or implement a method from a supertype
        @Override
        ^
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:44: error: method does not override or implement a method from a supertype
        @Override
        ^
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:49: error: method does not override or implement a method from a supertype
        @Override
        ^
8 errors

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed; see the compiler output below.
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:25: error: cannot access AbstractMojo
          mojo.setLog(log);
              ^
    class file for org.apache.maven.plugin.AbstractMojo not found
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:22: error: cannot find symbol
      void executeLoadsPluginHelpResource() throws MojoExecutionException {
                                                   ^
    symbol:   class MojoExecutionException
    location: class HelpMojoTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:35: error: cannot find symbol
      private static final class RecordingLog extends SystemStreamLog {
                                                      ^
    symbol:   class SystemStreamLog
    location: class HelpMojoTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:39: error: method does not override or implement a method from a supertype
          @Override
          ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:44: error: method does not override or implement a method from a supertype
          @Override
          ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:49: error: method does not override or implement a method from a supertype
          @Override
          ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:12: error: package org.apache.maven.plugin does not exist
  import org.apache.maven.plugin.MojoExecutionException;
                                ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:13: error: package org.apache.maven.plugin.logging does not exist
  import org.apache.maven.plugin.logging.SystemStreamLog;
                                        ^
  8 errors

* Try:
> Check your code and dependencies to fix the compilation error(s)
> Run with --scan
[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/build/reports/problems/problems-report.html
 to generate a Build Scan (Powered by Develocity).

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 902ms
10 actionable tasks: 1 executed, 9 up-to-date

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 2s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'javac compile' for 3.3.0 at 2026-06-07T18:50:22Z with exit code 1 after 3s
❌ ERROR: 'javac compile' for 3.3.0 failed with exit code 1.
FAILED[javac compile][3.3.0][env GVM_TCK_LV=3.3.0 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean compileTestJava -Pcoordinates=org.apache.maven.plugins:maven-surefire-plugin:3.0.0]
```

## javac compile on ubuntu-latest / JDK latest-ea / mode current-defaults

Failure kind: javac compile
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: current-defaults
Reproducer: `env GVM_TCK_LV=3.3.0 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean compileTestJava -Pcoordinates=org.apache.maven.plugins:maven-surefire-plugin:3.0.0`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/27101166930
Last 300 lines of the log (excerpt truncated to fit the GitHub issue size limit):
```console
^

> Task :compileTestJava FAILED
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:13: error: package org.apache.maven.plugin.logging does not exist
import org.apache.maven.plugin.logging.SystemStreamLog;
                                      ^
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:22: error: cannot find symbol
    void executeLoadsPluginHelpResource() throws MojoExecutionException {
                                                 ^
  symbol:   class MojoExecutionException
  location: class HelpMojoTest
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:35: error: cannot find symbol
    private static final class RecordingLog extends SystemStreamLog {
                                                    ^
  symbol:   class SystemStreamLog
  location: class HelpMojoTest
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:25: error: cannot access AbstractMojo
        mojo.setLog(log);
            ^
  class file for org.apache.maven.plugin.AbstractMojo not found
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:39: error: method does not override or implement a method from a supertype
        @Override
        ^
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:44: error: method does not override or implement a method from a supertype
        @Override
        ^
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:49: error: method does not override or implement a method from a supertype
        @Override
        ^
8 errors


FAILURE: Build failed with an exception.

* What went wrong:
[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/build/reports/problems/problems-report.html
Execution failed for task ':compileTestJava'.
> Compilation failed; see the compiler output below.
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:25: error: cannot access AbstractMojo
          mojo.setLog(log);
              ^
    class file for org.apache.maven.plugin.AbstractMojo not found
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:22: error: cannot find symbol
      void executeLoadsPluginHelpResource() throws MojoExecutionException {
                                                   ^
    symbol:   class MojoExecutionException
    location: class HelpMojoTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:35: error: cannot find symbol
      private static final class RecordingLog extends SystemStreamLog {
                                                      ^
    symbol:   class SystemStreamLog
    location: class HelpMojoTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:39: error: method does not override or implement a method from a supertype
          @Override
          ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:44: error: method does not override or implement a method from a supertype
          @Override
          ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:49: error: method does not override or implement a method from a supertype
          @Override
          ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:12: error: package org.apache.maven.plugin does not exist
  import org.apache.maven.plugin.MojoExecutionException;
                                ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:13: error: package org.apache.maven.plugin.logging does not exist
  import org.apache.maven.plugin.logging.SystemStreamLog;
                                        ^
  8 errors

* Try:
> Check your code and dependencies to fix the compilation error(s)
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 1s

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
12 actionable tasks: 2 executed, 10 up-to-date

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for org.apache.maven.plugins:maven-surefire-plugin:3.0.0 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 3s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 3.3.0 at 2026-06-07T18:50:17Z with exit code 1 after 4s
❌ ERROR: 'native-image run' for 3.3.0 failed with exit code 1.
SOFT-FAIL[native-image run][3.3.0][env GVM_TCK_LV=3.3.0 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean test -Pcoordinates=org.apache.maven.plugins:maven-surefire-plugin:3.0.0]
========================================================================================
 org.apache.maven.plugins:maven-surefire-plugin:3.0.0:3.3.0 stage javac compile
========================================================================================
Starting stage 'javac compile' for 3.3.0 at 2026-06-07T18:50:17Z
Command: env GVM_TCK_LV=3.3.0 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean compileTestJava -Pcoordinates=org.apache.maven.plugins:maven-surefire-plugin:3.0.0
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 750ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava FAILED
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :classes UP-TO-DATE

> Task :compileTestJava FAILED
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:12: error: package org.apache.maven.plugin does not exist
import org.apache.maven.plugin.MojoExecutionException;
                              ^
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:13: error: package org.apache.maven.plugin.logging does not exist
import org.apache.maven.plugin.logging.SystemStreamLog;
                                      ^
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:22: error: cannot find symbol
    void executeLoadsPluginHelpResource() throws MojoExecutionException {
                                                 ^
  symbol:   class MojoExecutionException
  location: class HelpMojoTest
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:35: error: cannot find symbol
    private static final class RecordingLog extends SystemStreamLog {
                                                    ^
  symbol:   class SystemStreamLog
  location: class HelpMojoTest
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:25: error: cannot access AbstractMojo
        mojo.setLog(log);
            ^
  class file for org.apache.maven.plugin.AbstractMojo not found
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:39: error: method does not override or implement a method from a supertype
        @Override
        ^
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:44: error: method does not override or implement a method from a supertype
        @Override
        ^
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:49: error: method does not override or implement a method from a supertype
        @Override
        ^
8 errors

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed; see the compiler output below.
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:25: error: cannot access AbstractMojo
          mojo.setLog(log);
              ^
    class file for org.apache.maven.plugin.AbstractMojo not found
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:22: error: cannot find symbol
      void executeLoadsPluginHelpResource() throws MojoExecutionException {
                                                   ^
    symbol:   class MojoExecutionException
    location: class HelpMojoTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:35: error: cannot find symbol
      private static final class RecordingLog extends SystemStreamLog {
                                                      ^
    symbol:   class SystemStreamLog
    location: class HelpMojoTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:39: error: method does not override or implement a method from a supertype
          @Override
          ^

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/build/reports/problems/problems-report.html
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:44: error: method does not override or implement a method from a supertype
          @Override
          ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:49: error: method does not override or implement a method from a supertype

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
10 actionable tasks: 1 executed, 9 up-to-date
          @Override
          ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:12: error: package org.apache.maven.plugin does not exist
  import org.apache.maven.plugin.MojoExecutionException;
                                ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:13: error: package org.apache.maven.plugin.logging does not exist
  import org.apache.maven.plugin.logging.SystemStreamLog;
                                        ^
  8 errors

* Try:
> Check your code and dependencies to fix the compilation error(s)
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 945ms

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 2s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'javac compile' for 3.3.0 at 2026-06-07T18:50:19Z with exit code 1 after 2s
❌ ERROR: 'javac compile' for 3.3.0 failed with exit code 1.
FAILED[javac compile][3.3.0][env GVM_TCK_LV=3.3.0 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean compileTestJava -Pcoordinates=org.apache.maven.plugins:maven-surefire-plugin:3.0.0]
```

## javac compile on ubuntu-latest / JDK latest-ea / mode future-defaults-all

Failure kind: javac compile
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: future-defaults-all
Reproducer: `env GVM_TCK_LV=3.3.0 GVM_TCK_NATIVE_IMAGE_MODE=future-defaults-all ./gradlew clean compileTestJava -Pcoordinates=org.apache.maven.plugins:maven-surefire-plugin:3.0.0`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/27101166930
Last 300 lines of the log (excerpt truncated to fit the GitHub issue size limit):
```console

                              ^
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:13: error: package org.apache.maven.plugin.logging does not exist
import org.apache.maven.plugin.logging.SystemStreamLog;
                                      ^
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:22: error: cannot find symbol
    void executeLoadsPluginHelpResource() throws MojoExecutionException {
                                                 ^
  symbol:   class MojoExecutionException
  location: class HelpMojoTest
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:35: error: cannot find symbol
    private static final class RecordingLog extends SystemStreamLog {
                                                    ^
  symbol:   class SystemStreamLog
  location: class HelpMojoTest
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:25: error: cannot access AbstractMojo
        mojo.setLog(log);
            ^
  class file for org.apache.maven.plugin.AbstractMojo not found
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:39: error: method does not override or implement a method from a supertype
        @Override
        ^
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:44: error: method does not override or implement a method from a supertype
        @Override
        ^

/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:49: error: method does not override or implement a method from a supertype
        @Override
        ^
8 errors
[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed; see the compiler output below.
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:25: error: cannot access AbstractMojo
          mojo.setLog(log);
              ^
    class file for org.apache.maven.plugin.AbstractMojo not found
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:22: error: cannot find symbol
      void executeLoadsPluginHelpResource() throws MojoExecutionException {
                                                   ^
    symbol:   class MojoExecutionException
    location: class HelpMojoTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:35: error: cannot find symbol
      private static final class RecordingLog extends SystemStreamLog {
                                                      ^
    symbol:   class SystemStreamLog
    location: class HelpMojoTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:39: error: method does not override or implement a method from a supertype
          @Override
          ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:44: error: method does not override or implement a method from a supertype
          @Override
          ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:49: error: method does not override or implement a method from a supertype
          @Override
          ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:12: error: package org.apache.maven.plugin does not exist
  import org.apache.maven.plugin.MojoExecutionException;
                                ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:13: error: package org.apache.maven.plugin.logging does not exist
  import org.apache.maven.plugin.logging.SystemStreamLog;
                                        ^
  8 errors

* Try:
> Check your code and dependencies to fix the compilation error(s)
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 1s

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
12 actionable tasks: 2 executed, 10 up-to-date

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for org.apache.maven.plugins:maven-surefire-plugin:3.0.0 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 2s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 3.3.0 at 2026-06-07T18:49:03Z with exit code 1 after 3s
❌ ERROR: 'native-image run' for 3.3.0 failed with exit code 1.
SOFT-FAIL[native-image run][3.3.0][env GVM_TCK_LV=3.3.0 GVM_TCK_NATIVE_IMAGE_MODE=future-defaults-all ./gradlew clean test -Pcoordinates=org.apache.maven.plugins:maven-surefire-plugin:3.0.0]
========================================================================================
 org.apache.maven.plugins:maven-surefire-plugin:3.0.0:3.3.0 stage javac compile
========================================================================================
Starting stage 'javac compile' for 3.3.0 at 2026-06-07T18:49:03Z
Command: env GVM_TCK_LV=3.3.0 GVM_TCK_NATIVE_IMAGE_MODE=future-defaults-all ./gradlew clean compileTestJava -Pcoordinates=org.apache.maven.plugins:maven-surefire-plugin:3.0.0
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 623ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava FAILED
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :tck-test-support:compileJava UP-TO-DATE

> Task :compileTestJava FAILED
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:12: error: package org.apache.maven.plugin does not exist
import org.apache.maven.plugin.MojoExecutionException;
                              ^
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:13: error: package org.apache.maven.plugin.logging does not exist
import org.apache.maven.plugin.logging.SystemStreamLog;
                                      ^
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:22: error: cannot find symbol
    void executeLoadsPluginHelpResource() throws MojoExecutionException {
                                                 ^
  symbol:   class MojoExecutionException
  location: class HelpMojoTest
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:35: error: cannot find symbol
    private static final class RecordingLog extends SystemStreamLog {
                                                    ^
  symbol:   class SystemStreamLog
  location: class HelpMojoTest
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:25: error: cannot access AbstractMojo
        mojo.setLog(log);
            ^
  class file for org.apache.maven.plugin.AbstractMojo not found
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:39: error: method does not override or implement a method from a supertype
        @Override
        ^
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:44: error: method does not override or implement a method from a supertype
        @Override

        ^
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:49: error: method does not override or implement a method from a supertype
        @Override
        ^
8 errors
[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed; see the compiler output below.
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:25: error: cannot access AbstractMojo
          mojo.setLog(log);
              ^
    class file for org.apache.maven.plugin.AbstractMojo not found
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:22: error: cannot find symbol
      void executeLoadsPluginHelpResource() throws MojoExecutionException {
                                                   ^
    symbol:   class MojoExecutionException
    location: class HelpMojoTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:35: error: cannot find symbol
      private static final class RecordingLog extends SystemStreamLog {
                                                      ^
    symbol:   class SystemStreamLog
    location: class HelpMojoTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:39: error: method does not override or implement a method from a supertype
          @Override
          ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:44: error: method does not override or implement a method from a supertype
          @Override
          ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:49: error: method does not override or implement a method from a supertype
          @Override
          ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:12: error: package org.apache.maven.plugin does not exist
  import org.apache.maven.plugin.MojoExecutionException;
                                ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.0.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java:13: error: package org.apache.maven.plugin.logging does not exist
  import org.apache.maven.plugin.logging.SystemStreamLog;
                                        ^
  8 errors

* Try:
> Check your code and dependencies to fix the compilation error(s)
> Run with --scan to generate a Build Scan (Powered by Develocity).

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 731ms
10 actionable tasks: 1 executed, 9 up-to-date

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 2s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'javac compile' for 3.3.0 at 2026-06-07T18:49:05Z with exit code 1 after 2s
❌ ERROR: 'javac compile' for 3.3.0 failed with exit code 1.
FAILED[javac compile][3.3.0][env GVM_TCK_LV=3.3.0 GVM_TCK_NATIVE_IMAGE_MODE=future-defaults-all ./gradlew clean compileTestJava -Pcoordinates=org.apache.maven.plugins:maven-surefire-plugin:3.0.0]
```

--- a/metadata/org.apache.maven.plugins/maven-surefire-plugin/3.3.0/reachability-metadata.json
+++ b/metadata/org.apache.maven.plugins/maven-surefire-plugin/3.3.0/reachability-metadata.json
@@ -1,0 +1,16 @@
+{
+  "resources": [
+    {
+      "condition": {
+        "typeReached": "org.apache.maven.plugins.maven_surefire_plugin.HelpMojo"
+      },
+      "glob": "META-INF/maven/org.apache.maven.plugins/maven-surefire-plugin/plugin-help.xml"
+    },
+    {
+      "condition": {
+        "typeReached": "org.apache.maven.plugins.maven_surefire_plugin.HelpMojo"
+      },
+      "glob": "META-INF/services/javax.xml.parsers.DocumentBuilderFactory"
+    }
+  ]
+}

--- a/metadata/org.apache.maven.plugins/maven-surefire-plugin/index.json
+++ b/metadata/org.apache.maven.plugins/maven-surefire-plugin/index.json
@@ -1,6 +1,15 @@
 [
   {
     "latest" : true,
+    "metadata-version" : "3.3.0",
+    "tested-versions" : [
+      "3.3.0"
+    ],
+    "allowed-packages" : [
+      "org.apache.maven.plugins.maven_surefire_plugin"
+    ]
+  },
+  {
     "metadata-version" : "3.0.0",
     "source-code-url" : "https://repo.maven.apache.org/maven2/org/apache/maven/plugins/maven-surefire-plugin/$version$/maven-surefire-plugin-$version$-sources.jar",
     "repository-url" : "https://github.com/apache/maven-surefire",

--- a/metadata/org.apache.maven.plugins/maven-surefire-plugin/index.json
+++ b/metadata/org.apache.maven.plugins/maven-surefire-plugin/index.json
@@ -2,6 +2,11 @@
   {
     "latest" : true,
     "metadata-version" : "3.3.0",
+    "source-code-url" : "https://repo.maven.apache.org/maven2/org/apache/maven/plugins/maven-surefire-plugin/$version$/maven-surefire-plugin-$version$-sources.jar",
+    "repository-url" : "https://github.com/apache/maven-surefire",
+    "test-code-url" : "https://repo.maven.apache.org/maven2/org/apache/maven/surefire/surefire/$version$/surefire-$version$-source-release.zip",
+    "documentation-url" : "https://repo.maven.apache.org/maven2/org/apache/maven/plugins/maven-surefire-plugin/$version$/maven-surefire-plugin-$version$-site-source.zip",
+    "description" : "The Maven Surefire Plugin runs a Maven project's unit tests during the test phase and reports the results. It integrates test providers such as JUnit, TestNG, and POJO-based tests with Maven's lifecycle, including options for filtering, forking, and failure handling.",
     "tested-versions" : [
       "3.3.0"
     ],

--- a/stats/org.apache.maven.plugins/maven-surefire-plugin/3.3.0/execution-metrics.json
+++ b/stats/org.apache.maven.plugins/maven-surefire-plugin/3.3.0/execution-metrics.json
@@ -1,0 +1,99 @@
+{
+  "fix_javac_fail:2026-06-07": {
+    "timestamp": "2026-06-07T20:36:48.936237Z",
+    "library": "org.apache.maven.plugins:maven-surefire-plugin:3.3.0",
+    "starting_commit": "c1dcda03ba397bc3f724c41eb635edc3dabf5ef7",
+    "ending_commit": "6ebdad34c0711faed34918e1b0b8a9eee8aedef4",
+    "previous_library": "org.apache.maven.plugins:maven-surefire-plugin:3.0.0",
+    "strategy_name": "javac_iterative_with_coverage_sources_pi_gpt-5.5",
+    "agent": "pi",
+    "model": "oca/gpt-5.5",
+    "status": "success",
+    "stats": {
+      "version": "3.3.0",
+      "dynamicAccess": {
+        "breakdown": {
+          "resources": {
+            "coverageRatio": 1.0,
+            "coveredCalls": 1,
+            "totalCalls": 1
+          }
+        },
+        "coverageRatio": 1.0,
+        "coveredCalls": 1,
+        "totalCalls": 1
+      },
+      "libraryCoverage": {
+        "instruction": {
+          "covered": 540,
+          "missed": 673,
+          "ratio": 0.445177,
+          "total": 1213
+        },
+        "line": {
+          "covered": 103,
+          "missed": 172,
+          "ratio": 0.374545,
+          "total": 275
+        },
+        "method": {
+          "covered": 14,
+          "missed": 83,
+          "ratio": 0.14433,
+          "total": 97
+        }
+      }
+    },
+    "previous_library_stats": {
+      "version": "3.0.0",
+      "dynamicAccess": {
+        "breakdown": {
+          "resources": {
+            "coverageRatio": 1.0,
+            "coveredCalls": 1,
+            "totalCalls": 1
+          }
+        },
+        "coverageRatio": 1.0,
+        "coveredCalls": 1,
+        "totalCalls": 1
+      },
+      "libraryCoverage": {
+        "instruction": {
+          "covered": 538,
+          "missed": 670,
+          "ratio": 0.445364,
+          "total": 1208
+        },
+        "line": {
+          "covered": 103,
+          "missed": 171,
+          "ratio": 0.375912,
+          "total": 274
+        },
+        "method": {
+          "covered": 14,
+          "missed": 83,
+          "ratio": 0.14433,
+          "total": 97
+        }
+      }
+    },
+    "metrics": {
+      "input_tokens_used": 26732,
+      "cached_input_tokens_used": 310784,
+      "output_tokens_used": 4382,
+      "iterations": 1,
+      "cost_usd": 0.2869,
+      "tested_library_loc": 103,
+      "metadata_entries": 2,
+      "previous_library_metadata_entries": 2,
+      "code_coverage_percent": 37.45,
+      "previous_library_coverage_percent": 37.59
+    },
+    "artifacts": {
+      "test_file": "tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.3.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java",
+      "metadata_file": "metadata/org.apache.maven.plugins/maven-surefire-plugin/3.3.0/reachability-metadata.json"
+    }
+  }
+}

--- a/stats/org.apache.maven.plugins/maven-surefire-plugin/3.3.0/stats.json
+++ b/stats/org.apache.maven.plugins/maven-surefire-plugin/3.3.0/stats.json
@@ -1,0 +1,37 @@
+{
+  "versions" : [ {
+    "version" : "3.3.0",
+    "dynamicAccess" : {
+      "breakdown" : {
+        "resources" : {
+          "coverageRatio" : 1.0,
+          "coveredCalls" : 1,
+          "totalCalls" : 1
+        }
+      },
+      "coverageRatio" : 1.0,
+      "coveredCalls" : 1,
+      "totalCalls" : 1
+    },
+    "libraryCoverage" : {
+      "instruction" : {
+        "covered" : 540,
+        "missed" : 673,
+        "ratio" : 0.445177,
+        "total" : 1213
+      },
+      "line" : {
+        "covered" : 103,
+        "missed" : 172,
+        "ratio" : 0.374545,
+        "total" : 275
+      },
+      "method" : {
+        "covered" : 14,
+        "missed" : 83,
+        "ratio" : 0.14433,
+        "total" : 97
+      }
+    }
+  } ]
+}

--- a/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.3.0/.gitignore
+++ b/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.3.0/.gitignore
@@ -1,0 +1,4 @@
+gradlew.bat
+gradlew
+gradle/
+build/

--- a/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.3.0/build.gradle
+++ b/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.3.0/build.gradle
@@ -12,6 +12,7 @@ String libraryVersion = tck.testedLibraryVersion.get()
 
 dependencies {
     testImplementation "org.apache.maven.plugins:maven-surefire-plugin:$libraryVersion"
+    testImplementation 'org.apache.maven:maven-plugin-api:3.6.3'
     testImplementation 'org.assertj:assertj-core:3.22.0'
 }
 

--- a/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.3.0/build.gradle
+++ b/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.3.0/build.gradle
@@ -1,0 +1,27 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+plugins {
+    id "org.graalvm.internal.tck"
+}
+
+String libraryVersion = tck.testedLibraryVersion.get()
+
+dependencies {
+    testImplementation "org.apache.maven.plugins:maven-surefire-plugin:$libraryVersion"
+    testImplementation 'org.assertj:assertj-core:3.22.0'
+}
+
+graalvmNative {
+    agent {
+        defaultMode = "conditional"
+        modes {
+            conditional {
+                userCodeFilterPath = "user-code-filter.json"
+            }
+        }
+    }
+}

--- a/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.3.0/gradle.properties
+++ b/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.3.0/gradle.properties
@@ -1,0 +1,6 @@
+# Optional explicit tested library coordinates (format: group:artifact:version).
+# If omitted, fallback resolution in org.graalvm.internal.tck.gradle still uses
+# environment variable GVM_TCK_LC and then this property.
+library.coordinates = org.apache.maven.plugins:maven-surefire-plugin:3.3.0
+library.version = 3.3.0
+metadata.dir = org.apache.maven.plugins/maven-surefire-plugin/3.3.0/

--- a/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.3.0/settings.gradle
+++ b/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.3.0/settings.gradle
@@ -1,0 +1,20 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+pluginManagement {
+    def tckPath = Objects.requireNonNullElse(
+            System.getenv("GVM_TCK_TCKDIR"),
+            "../../../../tck-build-logic"
+    )
+    includeBuild(tckPath)
+    repositories { mavenLocal(); gradlePluginPortal() }
+}
+
+plugins {
+    id "org.graalvm.internal.tck-settings" version "1.0.0-SNAPSHOT"
+}
+
+rootProject.name = 'org.apache.maven.plugins.maven-surefire-plugin_tests'

--- a/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.3.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java
+++ b/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.3.0/src/test/java/org_apache_maven_plugins/maven_surefire_plugin/HelpMojoTest.java
@@ -1,0 +1,54 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package org_apache_maven_plugins.maven_surefire_plugin;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugin.logging.SystemStreamLog;
+import org.apache.maven.plugins.maven_surefire_plugin.HelpMojo;
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class HelpMojoTest {
+
+    @Test
+    void executeLoadsPluginHelpResource() throws MojoExecutionException {
+        HelpMojo mojo = new HelpMojo();
+        RecordingLog log = new RecordingLog();
+        mojo.setLog(log);
+
+        mojo.execute();
+
+        assertThat(log.infoMessages)
+                .anySatisfy(message -> assertThat(message)
+                        .contains("Surefire")
+                        .contains("surefire:test"));
+    }
+
+    private static final class RecordingLog extends SystemStreamLog {
+
+        private final List<String> infoMessages = new ArrayList<>();
+
+        @Override
+        public boolean isInfoEnabled() {
+            return true;
+        }
+
+        @Override
+        public void info(CharSequence content) {
+            infoMessages.add(content.toString());
+        }
+
+        @Override
+        public void info(CharSequence content, Throwable error) {
+            info(content);
+        }
+    }
+}

--- a/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.3.0/user-code-filter.json
+++ b/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.3.0/user-code-filter.json
@@ -1,0 +1,16 @@
+{
+  "rules" : [
+    {
+      "excludeClasses" : "**"
+    },
+    {
+      "includeClasses" : "org.apache.maven.plugin.surefire.**"
+    },
+    {
+      "includeClasses" : "org.apache.maven.plugins.maven_surefire_plugin.**"
+    },
+    {
+      "includeClasses" : "org_apache_maven_plugins.maven_surefire_plugin.**"
+    }
+  ]
+}

--- a/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.3.0/user-code-filter.json
+++ b/tests/src/org.apache.maven.plugins/maven-surefire-plugin/3.3.0/user-code-filter.json
@@ -4,10 +4,10 @@
       "excludeClasses" : "**"
     },
     {
-      "includeClasses" : "org.apache.maven.plugin.surefire.**"
+      "includeClasses" : "org.apache.maven.plugins.maven_surefire_plugin.**"
     },
     {
-      "includeClasses" : "org.apache.maven.plugins.maven_surefire_plugin.**"
+      "includeClasses" : "org.apache.maven.plugin.surefire.**"
     },
     {
       "includeClasses" : "org_apache_maven_plugins.maven_surefire_plugin.**"

